torch/csrc/jit/ivalue.h

```diff
@@ -80,21 +80,24 @@ struct TORCH_API IValue {
       c10::raw::intrusive_ptr::incref(as_intrusive_ptr);
     }
   }
-  IValue(IValue&& rhs) noexcept : IValue() {
-    swap(rhs);
+  IValue(IValue&& rhs) noexcept
+  : payload(rhs.payload)
+  , tag(rhs.tag)
+  , is_intrusive_ptr(rhs.is_intrusive_ptr) {
+    rhs.clearToNone();
```

Comment:
Not sure about this one. Why?

Reply:
Yeah this seems unnecessary. We're carefully moving all arguments in the JIT, but this adds unnecessary zeroing on every move.

Reply:
The reasoning behind this is that if I move something, I want the old instance (say a Tensor) to be freed. If it's not set to None, the Tensor will be kept around until the old instance is destructed, which might not happen for quite some time.

Reply:
Sure, but you can always reset it to None yourself. It's unnecessary in most cases, and the move contract doesn't say anything about the state of the object moved from. The same issue happens with `std::vector`s, so it's nothing surprising.

Reply:
(note: this discussion is about the move assignment. The move constructor did already set rhs to None before this PR because it was swapped with an empty instance)

Reply:
Oh I actually introduced a bug here. The important thing I want to do here is to decrease the refcount here and potentially free the resources earlier. Setting it to None by itself doesn't do that yet. Setting it to None must then happen afterwards so the destructor doesn't do it again.

Reply:
On further thought, this *must* set rhs to None, otherwise the rhs destructor will double-free (or double-decrease) the refcount.

Reply:
But swap will do that, right?